secretstores/akeyless/akeyless.go

```diff
@@ -0,0 +1,866 @@
+				} else {
+					secretResultChannels <- secretResultCollection{name: item, value: value, err: nil}
+				}
+			}
+		}()
+	}
+
+	// close the channel when all goroutines are done
+	go func() {
+		wg.Wait()
+		close(secretResultChannels)
+	}()
+
+	// collect results and populate response
+	for result := range secretResultChannels {
+		if result.err != nil {
+			a.logger.Errorf("error getting secret '%s': %s. Skipping...", result.name, result.err.Error())
+			continue
+		}
+
+		response.Data[result.name] = map[string]string{result.name: result.value}
+	}
+
+	// Use the new BulkGetSecretResponse function to handle all secret types properly
+	// return BulkGetSecretResponse(ctx, itemsList.Items, a)
+	return response, nil
+}
+
+// Features returns the features available in this secret store.
+func (a *akeylessSecretStore) Features() []secretstores.Feature {
+	return []secretstores.Feature{}
+}
+
+// Close closes the secret store.
+func (a *akeylessSecretStore) Close() error {
+	if a.closeCh != nil {
+		close(a.closeCh)
+		a.wg.Wait()
+	}
+	return nil
+}
+
+// parseMetadata parses the metadata from the component configuration.
+func (a *akeylessSecretStore) parseMetadata(meta secretstores.Metadata) (*akeylessMetadata, error) {
+
+	a.logger.Debug("Parsing metadata...")
+	var m akeylessMetadata
+	err := kitmd.DecodeMetadata(meta.Properties, &m)
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate access ID
+	if m.AccessID == "" {
+		return nil, errors.New("accessId is required")
+	}
+
+	if !isValidAccessIdFormat(m.AccessID) {
+		return nil, errors.New("invalid accessId format, expected format is p-([A-Za-z0-9]{14}|[A-Za-z0-9]{12})")
+	}
+
+	// Set default gateway URL if not specified
+	if m.GatewayURL == "" {
+		a.logger.Infof("Gateway URL is not set, using default value %s...", PUBLIC_GATEWAY_URL)
+		m.GatewayURL = PUBLIC_GATEWAY_URL
+	} else {
+		_, err = url.ParseRequestURI(m.GatewayURL)
+		if err != nil {
+			return nil, fmt.Errorf("invalid gateway URL '%s': %w", m.GatewayURL, err)
+		}
+	}
+
+	// Trim trailing slash from gateway URL
+	m.GatewayURL = strings.TrimSuffix(m.GatewayURL, "/")
+
+	return &m, nil
+}
+
+func (a *akeylessSecretStore) GetSecretType(ctx context.Context, secretName string) (string, error) {
+
+	if err := a.ensureValidToken(ctx); err != nil {
+		return "", fmt.Errorf("failed to ensure valid token: %w", err)
+	}
+
+	describeItem := akeyless.NewDescribeItem(secretName)
+
+	a.mu.RLock()
+	token := a.token
+	a.mu.RUnlock()
+
+	describeItem.SetToken(token)
+	describeItemResp, httpResponse, err := a.v2.DescribeItem(ctx).Body(*describeItem).Execute()
+
+	if httpResponse != nil && httpResponse.StatusCode == http.StatusUnauthorized {
+		a.logger.Debug("received 401 unauthorized, re-authenticating...")
+		if err := a.ensureValidToken(ctx); err != nil {
+			return "", fmt.Errorf("failed to re-authenticate after 401: %w", err)
+		}
+
+		a.mu.RLock()
+		token = a.token
+		a.mu.RUnlock()
```

Comment:
oh, this is the dynamic token rotation? Seems like there are differences in the implementation of this across these funcs (ie looking down a func in GetSingleSecretValue it seems different with a helper). Can you create a helper used in all of them please? And ensure that everyone you need the token rotation logic that it is there 🙏